### PR TITLE
Add channel closes to activtiy

### LINKED ISSUE #564: Show force closes in activity

### PR DESCRIPTION
**Pull request by benthecarman**
Can be improved with more information once https://github.com/lightningdevkit/rust-lightning/issues/2343 is done.

For now this will at least tell us a force close happened in the activity.

Closes #564

### CODE REVIEW

File: mutiny-core/src/nodemanager.rs [hunk 1]
```diff
@@ -306,10 +307,50 @@ impl From<bdk::TransactionDetails> for TransactionDetails {
     }
 }
 
+/// Information about a channel that was closed.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ChannelClosure {
+    pub user_channel_id: Option<[u8; 16]>,
+    pub channel_id: Option<[u8; 32]>,
```
Comment (TonyGiorgio) [inline mutiny-core/src/nodemanager.rs:314]: Why an option when it can't be initialized without one?

File: mutiny-core/src/nodemanager.rs [hunk 2]
```diff
@@ -306,10 +307,50 @@ impl From<bdk::TransactionDetails> for TransactionDetails {
...
+    pub node_id: Option<PublicKey>,
+    pub reason: String,
+    pub timestamp: u64,
```
Comment (TonyGiorgio) [inline mutiny-core/src/nodemanager.rs:317]: this uses `timestamp` while others uses `last_updated`, not sure if you want to be consistent here.

File: mutiny-core/src/nodemanager.rs [hunk 3]
```diff
@@ -306,10 +307,50 @@ impl From<bdk::TransactionDetails> for TransactionDetails {
...
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for ChannelClosure {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.timestamp.cmp(&other.timestamp)
+    }
+}
```
Comment (TonyGiorgio) [inline mutiny-core/src/nodemanager.rs:347]: Should update the test for activity list sorting.

File: mutiny-wasm/src/lib.rs [hunk 4]
```diff
@@ -573,6 +573,33 @@ impl MutinyWallet {
         )?)
     }
 
+    /// Gets an invoice from the node manager.
+    /// This includes sent and received invoices.
```
Comment (TonyGiorgio) [inline mutiny-wasm/src/lib.rs:577]: Incorrect comment

File: mutiny-wasm/src/lib.rs [hunk 5]
```diff
@@ -573,6 +573,33 @@ impl MutinyWallet {
...
+        Ok(self
+            .inner
+            .node_manager
+            .get_channel_closure(u128::from_be_bytes(user_channel_id))
+            .await?
+            .into())
+    }
+
+    /// Gets an invoice from the node manager.
+    /// This includes sent and received invoices.
```
Comment (TonyGiorgio) [inline mutiny-wasm/src/lib.rs:593]: same

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

File: mutiny-core/src/nodemanager.rs (hunk 2 above)
Comment (benthecarman) [reply mutiny-core/src/nodemanager.rs:317]: Same here, backwards compatibility

File: mutiny-core/src/nodemanager.rs (hunk 3 above)
Comment (benthecarman) [reply mutiny-core/src/nodemanager.rs:347]: Added

**Review by TonyGiorgio — APPROVED** [review]

**Comment by TonyGiorgio** [comment]
I don't want a potential rebase with https://github.com/MutinyWallet/mutiny-node/pull/549 so please merge that first

**Pull request merged**